static/app/views/projectInstall/issueAlertNotificationOptions.tsx

```diff
@@ -0,0 +1,277 @@
+      );
+      if (
+        !organization.features.includes(
+          'messaging-integration-onboarding-project-creation'
+        ) ||
+        !shouldCreateRule ||
+        !isCreatingIntegrationNotification
+      ) {
+        return undefined;
+      }
+
+      let integrationAction;
+      switch (provider) {
+        case 'slack':
+          integrationAction = [
+            {
+              id: IssueAlertActionType.SLACK,
+              workspace: integration?.id,
+              channel: channel,
+            },
+          ];
+          break;
+        case 'discord':
+          integrationAction = [
+            {
+              id: IssueAlertActionType.DISCORD,
+              server: integration?.id,
+              channel_id: channel,
+            },
+          ];
+          break;
+        case 'msteams':
+          integrationAction = [
+            {
+              id: IssueAlertActionType.MS_TEAMS,
+              team: integration?.id,
+              channel: channel,
+            },
+          ];
+          break;
+        default:
+          return undefined;
+      }
+
+      return api.requestPromise(`/projects/${organization.slug}/${projectSlug}/rules/`, {
+        method: 'POST',
+        data: {
+          name,
+          conditions,
+          actions: integrationAction,
+          actionMatch,
+          frequency,
+        },
+      });
+    },
+    [
+      actions,
+      api,
+      provider,
+      integration,
+      channel,
+      organization.features,
+      organization.slug,
+    ]
+  );
+
+  return {
+    createNotificationAction,
+    notificationProps: {
+      actions,
+      provider,
+      integration,
+      channel,
+      setActions,
+      setProvider,
+      setIntegration,
+      setChannel,
+    },
+  };
+}
+
+export default function IssueAlertNotificationOptions(
+  notificationProps: IssueAlertNotificationProps
+) {
+  const organization = useOrganization();
+  const {actions, provider, setActions, setIntegration, setProvider} = notificationProps;
+
+  const messagingIntegrationsQuery = useApiQuery<OrganizationIntegration[]>(
+    [`/organizations/${organization.slug}/integrations/?integrationType=messaging`],
+    {staleTime: Infinity}
+  );
+
+  const providersToIntegrations = useMemo(() => {
+    const map: {[key: string]: OrganizationIntegration[]} = {};
+    if (messagingIntegrationsQuery.data) {
+      for (const i of messagingIntegrationsQuery.data) {
+        const providerSlug = i.provider.slug;
+        map[providerSlug] = map[providerSlug] ?? [];
+        map[providerSlug].push(i);
+      }
+    }
+    return map;
+  }, [messagingIntegrationsQuery.data]);
+
+  useEffect(() => {
+    const providerKeys = Object.keys(providersToIntegrations);
+    if (providerKeys.length > 0) {
+      const firstProvider = providerKeys[0];
+      setProvider(firstProvider);
+
+      const firstIntegration = providersToIntegrations[firstProvider][0];
+      setIntegration(firstIntegration);
```

Comment:
Do we actually need this state if we can just derive all of this?

Reply:
or is this just for setting up defaults?

Reply:
If it's for setting up defaults I think you could probably do this in the constructor functions of the set states in the hook. Assuming we move the api calls into that hook